pkg/controller/database/gitops_plan.go

```diff
@@ -0,0 +1,317 @@
+		}
+
+		if gvk.Group != "schemas.schemahero.io" || gvk.Version != "v1alpha2" || gvk.Kind != "Table" {
+			return nil
+		}
+
+		table := obj.(*schemasv1alpha2.Table)
+
+		if table.Namespace == "" {
+			table.Namespace = "default"
+		}
+
+		// We could (maybe should) deploy this and let the table reconciler run the plan...
+		// but that async nature will create a lot more code, and these are the same codebase/project.
+		// so let's just call over to the table controller directly for now
+
+		table.Spec.IsPlan = true
+		table.Name = fmt.Sprintf("%s-%s", table.Name, hash[0:7])
+		if table.Namespace == "" {
+			table.Namespace = "default"
+		}
+
+		tablePlan, err := r.getTablePlan(table, instance)
+		if err != nil {
+			return err
+		}
+
+		if len(plan) > 0 {
+			plan = plan + "\n"
+		}
+
+		plan = plan + tablePlan
+		return nil
+	})
+
+	if err != nil {
+		return "", errors.Wrap(err, "failed to walk file path")
+	}
+
+	return plan, nil
+}
+
+func (r *ReconcileDatabase) getTablePlan(table *schemasv1alpha2.Table, instance *databasesv1alpha2.Database) (string, error) {
+
+	// we need to deploy the table plan as a CR here because we need all of the logic that
+	// the reconcile loop in the schema package manages...
+
+	// this definitely makes this process a lot more complex than it really should be
+
+	existingObj := &schemasv1alpha2.Table{}
+	err := r.Get(context.Background(), types.NamespacedName{Name: table.Name, Namespace: table.Namespace}, existingObj)
+	if err != nil && !kuberneteserrors.IsNotFound(err) {
+		return "", errors.Wrap(err, "failed to look for existing object")
+	}
+
+	if kuberneteserrors.IsNotFound(err) {
+		// create
+		err = r.Create(context.Background(), table)
+		if err != nil {
+			return "", errors.Wrap(err, "failed to create table object")
+		}
+	} else {
+		// TODO how can we handle this?
+		// Delete and recreate?
+		// This won't happen in the "happy path" gitops flow, but certainly can happen in
+		// other use cases
+		return "", errors.New("cannot update existing table plan")
+	}
+
+	// Watch the object that we just created using an informer to get the plan
+	plan := ""
+
+	cfg, err := config.GetConfig()
+	if err != nil {
+		return "", errors.Wrap(err, "failed to get config")
+	}
+	schemasClient, err := schemasclientv1alpha2.NewForConfig(cfg)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to create schema client")
+	}
+
+	watchlist := cache.NewListWatchFromClient(schemasClient.RESTClient(), "tables", table.Namespace, fields.Everything())
+	resyncPeriod := 10 * time.Second
+	_, controller := cache.NewInformer(watchlist, &schemasv1alpha2.Table{}, resyncPeriod,
+		cache.ResourceEventHandlerFuncs{
+			UpdateFunc: func(oldObj interface{}, newObj interface{}) {
+				updatedTable := newObj.(*schemasv1alpha2.Table)
+				if updatedTable.Status.Plan != "" {
+					plan = updatedTable.Status.Plan
+				}
+			},
+		},
+	)
+
+	ctx := context.Background()
+	controller.Run(ctx.Done())
```

Comment:
im unfamiliar with informers but does this block? should it be in a goroutine or should the code below?